UnityProject/Assets/Scripts/Health/Living/PlayerHealth.cs

```diff
@@ -30,28 +30,52 @@ public class PlayerHealth : HealthBehaviour
 
 	public bool serverPlayerConscious { get; set; } = true; //Only used on the server
 
+	// JSON string for blood types and DNA.
+	[SyncVar(hook = "DNASync")]
+	private string DNABloodTypeJSON;
+	
+	// BloodType and DNA Data.
+	private DNAandBloodType DNABloodType;
+	
 	public override void OnStartClient()
 	{
 		playerNetworkActions = GetComponent<PlayerNetworkActions>();
 		playerMove = GetComponent<PlayerMove>();
 
 		PlayerScript playerScript = GetComponent<PlayerScript>();
-
+		
 		if (playerScript.JobType == JobType.NULL)
 		{
 			foreach (Transform t in transform)
 			{
 				t.gameObject.SetActive(false);
 			}
-
 			ConsciousState = ConsciousState.DEAD;
 
 			//Fixme: No more setting allowInputs on client:
 			playerMove.allowInput = false;
 		}
-
+		
+		// Gives DNA and BloodType to client.
+		DNABloodType = JsonUtility.FromJson<DNAandBloodType>(DNABloodTypeJSON);
+		
 		base.OnStartClient();
 	}
+	
+	public override void OnStartServer()
+	{
+		if (isServer)
+		{
+			DNABloodType = new DNAandBloodType();
+			DNABloodTypeJSON = JsonUtility.ToJson(DNABloodType);
+		}
+	}

```

Comment:
Don't forget to call any OnStartServer overrides in the base of this script (because it is derived from other behaviours, like HealthBehaviour.cs that might have an OnStartServer override on it that also needs to be called.)

Do that by adding the line: `base.OnStartServer(); somewhere in the method.